src/mappings/Transfers.ts

```diff
@@ -21,6 +25,61 @@ type TransferPayload = {
   assetId?: string;
 };
 
+export async function handleSwap(event: SubstrateEvent): Promise<void> {
+  const [from, to, path, amountIn, amountOut] = getEventData(event);
+
+  let element = await HistoryElement.get(eventId(event))
+
+  if (element !== undefined) {
+      // already processed swap previously
+      return;
+  }
+
+  let assetIdFee: string
+  let fee: string
+  let foundAssetTxFeePaid = event.block.events.find((e) => isAssetTxFeePaidEvent(eventRecordToSubstrateEvent(e)));
+  let swaps = event.block.events.filter((e) => isSwapExecutedEvent(eventRecordToSubstrateEvent(e)));
+  if (foundAssetTxFeePaid === undefined) {
+    assetIdFee = "native"
+    fee = calculateFeeAsString(event.extrinsic, from.toString())
+  } else {
+    const [who, actual_fee, tip, rawAssetIdFee] = getEventData(eventRecordToSubstrateEvent(foundAssetTxFeePaid))
+    assetIdFee = getAssetIdFromMultilocation(rawAssetIdFee)
+    fee = actual_fee.toString();
+    swaps = swaps.slice(1)
```

Comment:
We also need to check whether fee refund happened and take it into the account. As we found out swap for fee happens as first event and refund is last event (if it happens) and has opposite direction compared to the fee swap. So, I think we can determine the refunds by the amount in the fee paid event, amount in the fee swap event and the direction of the last swap.